Unbreak haskellPackages.HTF

shlevy opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Get `haskellPackages.HTF` building.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

shlevy committed:
haskellPackages.HTF: Fix testsuite and mark unbroken in configuration.

cdepillabout commented on pkgs/development/haskell-modules/hackage-packages.nix:
```diff
@@ -3486,6 +3486,28 @@ self: {
        broken = true;
      }) {};
 
+  "ConClusion" = callPackage
```
Could you rebase this branch on the `haskell-updates` branch?  We try to have all haskell-related changes go to that branch.

Also, if you rebase on `haskell-updates`, I'm guessing you can drop all the changes from `hackage-packages.nix`, since they should already be available on that branch?

shlevy replied:
Done.

cdepillabout commented on pkgs/development/haskell-modules/configuration-common.nix:
```diff
@@ -284,7 +284,11 @@ self: super: {
-  HTF = dontCheck super.HTF;
+  HTF = overrideCabal super.HTF (orig: {
+    preBuild = ''
+      patchShebangs --build scripts
```
Oh, and can you add a comment about why this is needed?  Or why it wasn't needed before?

It would be great if you could report this upstreamv as well and put a link to the upstream github issue here as a comment.

shlevy replied:
Added a comment.

There's not really a bug upstream. They use bash scripts, we just need to fixup for the sandbox. /bin/sh wouldn't be appropriate since they use bash features.

cdepillabout replied:
I was thinking they might be able to move to `#!/usr/bin/env bash`, which I guess we wouldn't have to fixup (and would be more portable to other distros)?

But this is no big deal.

shlevy replied:
Last I checked /usr/bin/env isn't in the sandbox?

sternenseemann replied:
Yeah, it is not.

cdepillabout replied:
Ah, that makes sense, sorry for the noise here then!

shlevy committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/ef920675c69d0a73d87ff602a3869faf0d5ae0d5.

cdepillabout commented:
This looks like it is now building, thanks!
